src/vcs/git/reference/glob.rs

```
@@ -0,0 +1,127 @@
+        self.with_namespace_glob(namespace, repo)
+    }
+
+    fn with_namespace_glob<'a>(
+        &self,
+        namespace: Namespace,
+        repo: &RepositoryRef<'a>,
+    ) -> Result<References<'a>, error::Error> {
+        let mut namespace_glob = "".to_string();
+        for n in namespace.values {
+            namespace_glob.push_str(&format!("refs/namespaces/{}/", n));
+        }
+
+        Ok(match self {
+            Self::Branch => {
+                let remotes = repo.repo_ref.references_glob(&format!(
+                    "{}{}",
+                    namespace_glob,
+                    Self::RemoteBranch { remote: None }.to_string()
+                ))?;
+
+                let locals = repo.repo_ref.references_glob(&format!(
+                    "{}{}",
+                    namespace_glob,
+                    &Self::LocalBranch.to_string()
+                ))?;
+                References {
+                    inner: vec![remotes, locals],
+                }
+            },
+            other => References {
+                inner: vec![repo.repo_ref.references_glob(&format!(
+                    "{}{}",
+                    namespace_glob,
+                    other.to_string()
+                ))?],
+            },
+        })
+    }
+}
+
+impl fmt::Display for RefGlob {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Tag => write!(f, "refs/tags/*"),
+            Self::LocalBranch => write!(f, "refs/heads/*"),
+            Self::RemoteBranch { remote } => {
+                write!(f, "refs/remotes/")?;
+                match remote {
+                    None => write!(f, "**/*"),
+                    Some(remote) => write!(f, "{}/*", remote),
+                }
+            },
+            // Note: the glob below would be used, but libgit doesn't care for union globs.
+            // write!(f, "refs/{{remotes/**/*,heads/*}}")
+            Self::Branch => panic!(
```

wait what? why is this panicking in `Display`?

---

So it should be the above in the comment, but that glob doesn't work. It'll just return no references. So I'm not sure what's worse: 1) silent failure 2) fail as loud as possible. It's an internal module so the Display shouldn't be called on it. Maybe I should mark everything as `pub(super)`?